rest-service/manager_rest/test/endpoints/test_searches.py

```diff
@@ -0,0 +1,89 @@
+        self.put_deployment_with_labels(self.LABELS_2)
+        deployments = self.client.deployments.list(
+            filter_rules=[FilterRule('KEy1', ['VaL1'], 'any_of', 'label')])
+        self.assertEqual(len(deployments), 2)
+        self.assert_metadata_filtered(deployments, 0)
+
+    def test_list_deployments_with_filter_rules_and_filter_id(self):
+        self.put_deployment_with_labels(self.LABELS)
+        self.put_deployment_with_labels(self.LABELS_2)
+        dep3 = self.put_deployment_with_labels(self.LABELS_3)
+        self._test_list_resources_with_filter_rules_and_filter_id(
+            self.client.deployments_filters, self.client.deployments, dep3)
+
+    def test_list_blueprints_with_filter_rules(self):
+        for i in range(1, 3):
+            bp_file_name = 'blueprint_with_labels_{0}.yaml'.format(i)
+            bp_id = 'blueprint_{0}'.format(i)
+            self.put_blueprint(blueprint_id=bp_id,
+                               blueprint_file_name=bp_file_name)
+        all_blueprints = self.client.blueprints.list(
+            filter_rules=[
+                FilterRule('bp_key1', ['bp_key1_val1'], 'any_of', 'label')])
+        second_blueprint = self.client.blueprints.list(
+            filter_rules=[
+                FilterRule('bp_key2', ['bp_2_val1'], 'any_of', 'label'),
+                FilterRule('bp_key1', [], 'is_not_null', 'label')])
+        self.assertEqual(len(all_blueprints), 2)
+        self.assert_metadata_filtered(all_blueprints, 0)
+        self.assertEqual(len(second_blueprint), 1)
+        self.assert_metadata_filtered(second_blueprint, 1)
+
+    def test_list_blueprints_with_filter_rules_and_filter_id(self):
+        self.put_blueprint_with_labels(self.LABELS, blueprint_id='bp1')
+        self.put_blueprint_with_labels(self.LABELS_2, blueprint_id='bp2')
+        bp3 = self.put_blueprint_with_labels(self.LABELS_3,
+                                             blueprint_id='bp3')
+        self._test_list_resources_with_filter_rules_and_filter_id(
+            self.client.blueprints_filters, self.client.blueprints, bp3)
+
+    def test_list_deployments_with_duplicate_filter_rules(self):
+        dep1 = self.put_deployment_with_labels(self.LABELS)
+        self.put_deployment_with_labels(self.LABELS_2)
+        self.create_filter(self.client.deployments_filters, self.FILTER_ID,
+                           self.FILTER_RULES)
+        deployments = self.client.deployments.list(
+            filter_rules=self.FILTER_RULES, filter_id=self.FILTER_ID)
+        self.assertEqual(len(deployments), 1)
+        self.assertEqual(deployments[0], dep1)
+        self.assert_metadata_filtered(deployments, 1)
+
+    def _test_list_resources_with_filter_rules_and_filter_id(self,
+                                                             filters_client,
+                                                             resource_client,
+                                                             cmp_resource):
```

Comment:
please rename `cmp_resource`. I can't understand what cmp stands for

Reply:
`cmp_resource` stands for `compared_resorce`.